e2e/_suites/fleet/features/fleet_mode_agent.feature

```diff
@@ -40,6 +40,17 @@ Examples:
 | centos |
 | debian |
 
+@upgrade-agent
+Scenario Outline: Upgrading the installed <os> agent
+  Given a "<os>" agent "stale" is deployed to Fleet with "tar" installer
+    And certs for "<os>" are installed
+  When agent is upgraded to version "latest"
+  Then process waits for "2m"
```

Comment:
when you are ready... a proper polling state is likely quicker and more stable than the long 2 min wait here.